tests/framework/enrolment-providers/class-sensei-test-enrolment-provider-always-provides.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * File containing the class Sensei_Test_Access_Provider_Always_Provides.
+ *
+ * @package sensei-tests
+ */
+
+/**
+ * Class Sensei_Test_Enrolment_Provider_Always_Provides.
+ *
+ * Used in testing. Always provides enrolment.
+ */
+class Sensei_Test_Enrolment_Provider_Always_Provides implements Sensei_Course_Enrolment_Provider_Interface {
+	public static function get_id() {
+		return 'always-provides';
+	}
+
+	public function handles_enrolment( $course_id ) {
+		return true;
+	}
+
+	public function is_enroled( $user_id, $course_id ) {
+		return true;
+	}
+
+	public static function get_version() {
+		return 1;
```

Comment:
What would a string do? What bugs do you think it might mask? Versions are always paired with the provider ID before hashing, so I don't think it should affect hashes.

Reply:
> What would a string do?

I was thinking that the version numbers for providers in the real world would be strings. I guess now that I think of it, integers would make just as much sense. So I'm fine with leaving it as integers.

> What bugs do you think it might mask?

Nothing specific comes to mind, for the reason you mentioned. But it does seem like the sort of thing that could conceivably crop up in the future. For example, if there is some code that sets up an ordered array of version numbers for some reason, and the order is wrong, our tests won't pick up on it. Or if something grabs the version number for the wrong provider, the tests will not fail.

It's a hypothetical class of problems that may never come up. But making the version numbers unique seems like a harmless way to prevent against them.